servicetalk-loadbalancer-experimental/src/main/java/io/servicetalk/loadbalancer/HostSelector.java

```diff
@@ -71,10 +72,11 @@ Single<C> selectConnection(Predicate<C> selector, @Nullable ContextMap context,
     boolean isHealthy();
 
     /**
-     * The size of the host candidate pool for this host selector.
+     * The set of hosts this selector will pick from.
      * <p>
-     * Note that this is primarily for observability purposes.
-     * @return the size of the host candidate pool for this host selector.
+     * Note that this may differ from the hosts advertised by the {@link io.servicetalk.client.api.ServiceDiscoverer}
+     * due to various filtering mechanisms.
+     * @return the set of hosts this selector will pick from.
      */
-    int hostSetSize();
+    Collection<? extends Host<ResolvedAddress, C>> hosts();
```

Comment:
Should we use `? extends LoadBalancerObserver.Host` here to avoid exposing methods that can modify internal `Host` state? Even though the `HostSelector` currently an internal interface, it may help to avoid unintentional use of this method.